### ISSUE TITLE
How to embed sequence in batch using ESM-C via Forge?

### ISSUE THREAD

**Issue by lzygitk7** [evolutionaryscale/esm#194]
Dear Author,

Thank you for your excellent work. I have a question about using ESM-C via Forge to embed sequences in batches.
The ESM-C model supports input shapes like [batch_size, sequence_length]. However, in practice, it seems ESM-C Forge processes inputs one at a time, taking about 5 seconds each. (Also the ESM3ForgeInferenceClient object lacks a tokenizer.)
Is there a solution for batch processing? Thank you in advance.

**Comment by lzygitk7** [comment]
And I have another question. 
I found the output has several attributes like embeddings/hidden_states...

![Image](https://github.com/user-attachments/assets/5364c1d3-7297-436d-9525-ecf928979e31)
According to the tutorial. The embedding is extracted from the hidden_state. So what are the other results meaning?
Using the 600M model via ESMforgeclient to analyze the same dataset in the tutorial, the result of PCA analysis for different layer representations shows unstable fluctuation.

![Image](https://github.com/user-attachments/assets/eece3e82-d288-4b80-9cac-9231b8adcab0)

**Comment by thomas-a-neil** [comment]
Thanks for your question.

It seems you've already found the [tutorial notebook](https://github.com/evolutionaryscale/esm/blob/main/cookbook/tutorials/2_embed.ipynb) that is relevant for batch processing. We do all batch processing on our server side, so you don't need to batch. Rather, you can submit asynchronous calls to the server and it will do batch processing for you. 

Regarding variance in PCA across layers:  We have seen that different layers represent different information about the sequence. Layer 0, for example, is merely the encoder layer, so we don't expect much structural information to be present at that layer

Regarding the outputs:
* `embeddings` correspond to the last layer embedding
* `hidden states` include all layer embeddings, as you mentioned
* `logits` are the sequence logits
* `residue_annotation_logits` - this is currently a placeholder field that isn't supported

**Comment by lzygitk7** [comment]
> Thanks for your question.
> 
> It seems you've already found the [tutorial notebook](https://github.com/evolutionaryscale/esm/blob/main/cookbook/tutorials/2_embed.ipynb) that is relevant for batch processing. We do all batch processing on our server side, so you don't need to batch. Rather, you can submit asynchronous calls to the server and it will do batch processing for you.
> 
> Regarding variance in PCA across layers: We have seen that different layers represent different information about the sequence. Layer 0, for example, is merely the encoder layer, so we don't expect much structural information to be present at that layer
> 
> Regarding the outputs:
> 
> * `embeddings` correspond to the last layer embedding
> * `hidden states` include all layer embeddings, as you mentioned
> * `logits` are the sequence logits
> * `residue_annotation_logits` - this is currently a placeholder field that isn't supported

Many thanks for your reply!
